core/unit_test/TestTeam.hpp

```diff
@@ -304,41 +348,78 @@ class TestReduceTeam {
   TestReduceTeam(const size_type &nwork) { run_test(nwork); }
 
   void run_test(const size_type &nwork) {
-    using functor_type =
-        Test::ReduceTeamFunctor<ScalarType, execution_space, ScheduleType>;
-    using value_type = typename functor_type::value_type;
-    using result_type =
-        Kokkos::View<value_type, Kokkos::HostSpace, Kokkos::MemoryUnmanaged>;
-
     enum { Count = 3 };
     enum { Repeat = 100 };
 
-    value_type result[Repeat];
-
     const uint64_t nw   = nwork;
     const uint64_t nsum = nw % 2 ? nw * ((nw + 1) / 2) : (nw / 2) * (nw + 1);
 
     policy_type team_exec(nw, 1);
 
-    const unsigned team_size = team_exec.team_size_recommended(
-        functor_type(nwork), Kokkos::ParallelReduceTag());
-    const unsigned league_size = (nwork + team_size - 1) / team_size;
+    // non-array reduction
+    {
+      using functor_type =
+          Test::ReduceTeamFunctor<ScalarType, execution_space, ScheduleType>;
+      using value_type = typename functor_type::value_type;
+      using result_type =
+          Kokkos::View<value_type, Kokkos::HostSpace, Kokkos::MemoryUnmanaged>;
 
-    team_exec = policy_type(league_size, team_size);
+      value_type result[Repeat];
 
-    for (unsigned i = 0; i < Repeat; ++i) {
-      result_type tmp(&result[i]);
-      Kokkos::parallel_reduce(team_exec, functor_type(nwork), tmp);
+      const unsigned team_size = team_exec.team_size_recommended(
+          functor_type(nwork), Kokkos::ParallelReduceTag());
+      const unsigned league_size = (nwork + team_size - 1) / team_size;
+
+      team_exec = policy_type(league_size, team_size);
+
+      for (unsigned i = 0; i < Repeat; ++i) {
+        result_type tmp(&result[i]);
+        Kokkos::parallel_reduce(team_exec, functor_type(nwork), tmp);
+      }
+
+      execution_space().fence();
+
+      for (unsigned i = 0; i < Repeat; ++i) {
+        for (unsigned j = 0; j < Count; ++j) {
+          const uint64_t correct = (j == 0) ? nw : nsum;
+          ASSERT_EQ((ScalarType)correct, result[i].value[j]);
+        }
+      }
     }
 
-    execution_space().fence();
+// FIXME_MSVC FIXME_32BIT Test is known to fail
```

Comment:
Fail or not compile?
If that is fail why are we not skipping instead of guarding the whole thing?

Reply:
Mostly, because the output would be misleading since only have the test is skipped.

Reply:
Can/should these be split into two unit tests?  (Haven't looked into details)